### PR TITLE
Rewrite soft

### PR DESCRIPTION
**Pull request by MikailBag**

### CODE REVIEW

File: src/configure-toolchains/src/trace.rs [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+}
+
+fn run_under_trace(
+    script_path: &Path,
+    data_path: &Path,
+    detect_out: &DetectScriptOutput,
+    work_dir: &Path,
+    tracer: Tracer,
+) -> anyhow::Result<Vec<u8>> {
+    println!("running in {}", work_dir.display());
```
Comment (pavel-the-best) [inline src/configure-toolchains/src/trace.rs]: Probably message should start with capital letter

File: src/configure-toolchains/src/trace.rs [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    let log_out_file = work_dir.join("__jjs_trace.json");
+    let data_path = data_path.canonicalize().context("data dir not exists")?;
+    println!("script will use data from {}", data_path.display());
```
Comment (pavel-the-best) [inline src/configure-toolchains/src/trace.rs]: -//-

File: src/configure-toolchains/src/trace.rs [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        Tracer::Lxtrace
+    };
+    let scripts: anyhow::Result<Vec<_>> = tpl
+        .dir
+        .join("use")
+        .read_dir()?
+        .map(|item| item.map_err(|err| anyhow::Error::new(err).context("failed to read script")))
+        .collect();
+    for script in scripts? {
+        println!("running {}", script.path().display());
```
Comment (pavel-the-best) [inline src/configure-toolchains/src/trace.rs]: -//-

**Comment by MikailBag** [comment]
@pavel-the-best 
I will left it for other PR.
Please note that primary stragegy is still tracing.

**Pull request merged**